src/GraphQL/Types/EnumerationGraphType.cs

```diff
@@ -33,7 +33,8 @@ public void AddValue(EnumValue value)
         public override object Coerce(object value)
         {
             var found = Values.FirstOrDefault(v => v.Value.Equals(value));
-            return found != null ? found.Name : null;
+            if (found != null) return found.Name;
+            return Values.FirstOrDefault(v => v.Name.Equals(value))?.Value;
```

Comment:
If memory serves me right, the original code works when mapping out to JSON, but it does not work for inbound conversion of enums. I'll add a test.